pkg/apis/greenhouse/v1alpha1/team_types.go

```diff
@@ -47,3 +47,9 @@ type TeamList struct {
 func init() {
 	SchemeBuilder.Register(&Team{}, &TeamList{})
 }
+
+func (o *Team) GetConditions() StatusConditions {
+	return StatusConditions{} // OrganizationStatus is an empty struct so we don't need to get anything
+}
+
+func (o *Team) SetCondition(Condition) {} // OrganizationStatus is an empty struct so we don't need to set anything
```

Comment:
The lifecycle pkg will set default `Delete` condition on the resource it is reconciling (Team)

We need this for `EnsureDeleted` otherwise there will be an endless reconcile loop and finalizers will not be removed